moviebot/controller/controller_flask_socket.py

```diff
@@ -0,0 +1,155 @@
+"""This file contains the Controller class which controls the flow of the
+conversation while the user interacts with the agent using Flask Socket IO."""
+
+import logging
+from dataclasses import asdict
+from typing import Any, Dict, Type
+
+from flask import request, session
+from flask_socketio import emit
+
+from dialoguekit.core import Utterance
+from dialoguekit.platforms.flask_socket_platform import (
+    ChatNamespace as DKChatNamespace,
+)
+from dialoguekit.platforms.flask_socket_platform import FlaskSocketPlatform
+from moviebot.agent.agent import MovieBotAgent
+from moviebot.controller.controller import Controller
+from moviebot.controller.http_data_formatter import Message, Response
+from moviebot.database.db_users import UserDB
+
+logger = logging.getLogger(__name__)
+
+
+class ControllerFlaskSocket(Controller, FlaskSocketPlatform):
+    def __init__(
+        self,
+        agent_class: Type[MovieBotAgent],
+        agent_args: Dict[str, Any] = {},
+    ) -> None:
+        """Initializes structs for Controller and sends the get_started button
+        to the client.
+
+        Args:
+            agent_class: The class of the agent.
+            agent_args: The arguments to pass to the agent. Defaults to empty
+              dict.
+        """
+        super().__init__(agent_class, agent_args)
+
+    def start(self, host: str = "127.0.0.1", port: str = "5000") -> None:
+        """Starts the platform.
+
+        Args:
+            host: Hostname. Defaults to 127.0.0.1.
+            port: Port. Defaults to 5000.
+        """
+        self.socketio.on_namespace(ChatNamespace("/", self))
+        self.socketio.run(self.app, host=host, port=port)
+
+    def display_agent_utterance(
```

Comment:
Seems this method is already defined in `FlaskSocketPlatform`.

Reply:
That is correct however here we use different classes `Message` and `Response` that are tailored for our use case.